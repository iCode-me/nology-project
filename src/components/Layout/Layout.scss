@use "../../styles/variables" as *;

.layout {
  min-height: 100vh;
  width: 100vw;
  background-color: #2c2a30;

  &__main {
    padding: 1rem;
  }
}

@media screen and (min-width: $tablet-breakpoint) {
  .layout {
    &__main {
      padding: 2rem;
    }
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .layout {
    &__main {
      padding: 2rem 4rem;
    }
  }
}
