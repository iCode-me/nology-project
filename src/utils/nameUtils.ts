export const formatName = (firstName: string, lastName: string): string => {
  return `${firstName} ${lastName}`;
};

// export const formatNameArray = (names: string[]): string => {
//   return names.join(" ");
// };
