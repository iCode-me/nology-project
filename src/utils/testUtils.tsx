import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};
