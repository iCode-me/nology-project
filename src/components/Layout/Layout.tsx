import "./Layout.scss";

type LayoutProps = {
  exampleProp?: boolean;
};

const ROOT_CLASSNAME = "layout";

// exampleProp is a placeholder for some layout specific property that may differ from page to page; enableSidebar, showNavigation etc.
// this can be useful for user journeys where you don't want your user to be able to navigate away as easily
const Layout: React.FC<LayoutProps> = ({ exampleProp = false, children }) => {
  return (
    <div className={ROOT_CLASSNAME}>
      {exampleProp && <span>exampleProp is true</span>}
      <main className={`${ROOT_CLASSNAME}__main`}>{children}</main>
    </div>
  );
};

export default Layout;
