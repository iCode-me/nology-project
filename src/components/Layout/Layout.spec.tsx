import { customRender } from "../../utils/testUtils";
import Layout from "./Layout";

export default describe("Layout", () => {
  it("Renders without children", () => {
    const { container } = customRender(<Layout />);

    expect(container).toMatchSnapshot();
  });

  it("Renders with children", () => {
    const { container } = customRender(
      <Layout>
        <span>Child element</span>
      </Layout>
    );

    expect(container).toMatchSnapshot();
  });

  it("Renders with exampleProp enabled, no children", () => {
    const { container } = customRender(<Layout exampleProp />);

    expect(container).toMatchSnapshot();
  });

  it("Renders with exampleProp enabled, children", () => {
    const { container } = customRender(
      <Layout exampleProp>
        <span>Child element</span>
      </Layout>
    );

    expect(container).toMatchSnapshot();
  });
});
