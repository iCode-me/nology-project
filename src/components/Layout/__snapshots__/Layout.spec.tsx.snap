// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Renders with children 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <span>
        Child element
      </span>
    </main>
  </div>
</div>
`;

exports[`Layout Renders with exampleProp enabled, children 1`] = `
<div>
  <div
    class="layout"
  >
    <span>
      exampleProp is true
    </span>
    <main
      class="layout__main"
    >
      <span>
        Child element
      </span>
    </main>
  </div>
</div>
`;

exports[`Layout Renders with exampleProp enabled, no children 1`] = `
<div>
  <div
    class="layout"
  >
    <span>
      exampleProp is true
    </span>
    <main
      class="layout__main"
    />
  </div>
</div>
`;

exports[`Layout Renders without children 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    />
  </div>
</div>
`;
